flow/core/kernel/scenario/base.py

```diff
@@ -251,22 +253,97 @@ def gen_even_start_pos(self, initial_config, num_vehicles, **kwargs):
             see flow/core/params.py
         num_vehicles : int
             number of vehicles to be placed on the network
-        kwargs : dict
-            extra components, usually defined during reset to overwrite initial
-            config parameters
 
         Returns
         -------
         list of tuple (float, float)
             list of start positions [(edge0, pos0), (edge1, pos1), ...]
         list of int
             list of start lanes
-        list of float
-            list of start speeds
         """
-        raise NotImplementedError
-
-    def gen_random_start_pos(self, initial_config, num_vehicles, **kwargs):
+        (x0, min_gap, bunching, lanes_distr, available_length,
+         available_edges, initial_config) = \
+            self._get_start_pos_util(initial_config, num_vehicles)
+
+        # return an empty list of starting positions and lanes if there are no
+        # vehicles to be placed
+        if num_vehicles == 0:
+            return [], []
+
+        increment = available_length / num_vehicles
+
+        # if not all lanes are equal, then we must ensure that vehicles are in
```

Comment:
This comment doesn't seem correctly worded, I'm not sure what it's trying to get across.